src/test/java/com/elasticbox/jenkins/SlaveProvisionTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * ElasticBox Confidential
+ * Copyright (c) 2015 All Right Reserved, ElasticBox Inc.
+ *
+ * NOTICE:  All information contained herein is, and remains the property
+ * of ElasticBox. The intellectual and technical concepts contained herein are
+ * proprietary and may be covered by U.S. and Foreign Patents, patents in process,
+ * and are protected by trade secret or copyright law. Dissemination of this
+ * information or reproduction of this material is strictly forbidden unless prior
+ * written permission is obtained from ElasticBox.
+ */
+
+package com.elasticbox.jenkins;
+
+import com.elasticbox.Client;
+import com.elasticbox.jenkins.ElasticBoxCloud;
+import com.elasticbox.jenkins.ElasticBoxExecutor;
+import com.elasticbox.jenkins.ElasticBoxSlave;
+import com.elasticbox.jenkins.ElasticBoxSlaveHandler;
+import com.elasticbox.jenkins.SlaveConfiguration;
+import com.elasticbox.jenkins.tests.SlaveProvisionTestBase;
+import com.elasticbox.jenkins.tests.TestUtils;
+import com.elasticbox.jenkins.util.Condition;
+import hudson.ExtensionList;
+import hudson.model.Node;
+import net.sf.json.JSONArray;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ *
+ * @author Phong Nguyen Le
```

Comment:
😄, well, he was the original author, but since we don't typically add the author lately, I'm removing it!